src/OpenTelemetry.Api/Trace/Status.cs

```diff
@@ -142,7 +142,13 @@ namespace OpenTelemetry.Trace
         /// </summary>
         public static readonly Status DataLoss = new Status(StatusCanonicalCode.DataLoss);
 
-        internal Status(StatusCanonicalCode statusCanonicalCode, string description = null)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Status"/> struct.
+        /// May not want to make this public - just doing so for conversation.
+        /// </summary>
+        /// <param name="statusCanonicalCode">Canonical code.</param>
+        /// <param name="description">Description.</param>
+        public Status(StatusCanonicalCode statusCanonicalCode, string description = null)
```

Comment:
I'm guessing this is not the approach we want to take. There’s a 1:1 between gRPC status codes and OpenTelemetry codes see [rpc spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md#status). Instead of exposing the constructor perhaps it makes sense to have a static method like `public static Status StatusFromCanonicalCode(StatusCanonicalCode statusCanonicalCode, string description = null)`.

Thoughts?